src/keylessBackup/KeylessBackupProgress.tsx

```diff
@@ -24,21 +29,77 @@ const TAG = 'keylessBackup/KeylessBackupProgress'
 function KeylessBackupProgress({
   route,
 }: NativeStackScreenProps<StackParamList, Screens.KeylessBackupProgress>) {
-  const keylessBackupStatus = useSelector(keylessBackupStatusSelector)
-
-  const { t } = useTranslation()
-
   // Disable back button on Android
   useEffect(() => {
     const backPressListener = () => true
     BackHandler.addEventListener('hardwareBackPress', backPressListener)
     return () => BackHandler.removeEventListener('hardwareBackPress', backPressListener)
   }, [])
 
-  // TODO(ACT-781): Implement Restore flow designs
   if (route.params.keylessBackupFlow === KeylessBackupFlow.Restore) {
-    return <></>
+    return <Restore />
+  } else {
+    return <Setup />
   }
+}
+
+function Restore() {
+  const keylessBackupStatus = useSelector(keylessBackupStatusSelector)
+  const localCurrencyCode = useLocalCurrencyCode()
+
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+
+  switch (keylessBackupStatus) {
+    case KeylessBackupStatus.InProgress: {
+      return InProgress(t('keylessBackupStatus.restore.inProgress.title'))
+    }
+    case KeylessBackupStatus.RestoreZeroBalance: {
+      return (
+        <SafeAreaView>
+          <Dialog
+            title={
+              <Trans i18nKey="importExistingKey.emptyWalletDialog.title">
+                <TokenDisplay
+                  localAmount={{
+                    value: new BigNumber(0),
+                    currencyCode: localCurrencyCode,
+                    exchangeRate: '1',
+                  }}
+                  showLocalAmount={true}
+                  amount={new BigNumber(0)}
+                />
+              </Trans>
+            }
+            isVisible={true}
+            actionText={t('importExistingKey.emptyWalletDialog.action')}
+            actionPress={() => {
+              ValoraAnalytics.track(KeylessBackupEvents.cab_restore_zero_balance_accept)
+              dispatch(keylessBackupAcceptZeroBalance())
+            }}
+            secondaryActionText={t('goBack')}
+            secondaryActionPress={() => {
+              ValoraAnalytics.track(KeylessBackupEvents.cab_restore_zero_balance_bail)
+              dispatch(keylessBackupBail())
+            }}
+            testID="ConfirmUseAccountDialog"
+          >
+            {t('importExistingKey.emptyWalletDialog.description')}
+          </Dialog>
+        </SafeAreaView>
+      )
+    }
+    // TODO(ACT-781): Implement Success screen
+    // TODO(ACT-780): Implement Failure screens
+    default:
+      Logger.error(TAG, `Got unexpected keyless backup status: ${keylessBackupStatus}`)
```

Comment:
NotStarted will fall here right? Any reason why this is handled differently for setup and restore?

Reply:
I believe that the NotStarted state cannot be reached on this screen with the restore flow OR the setup flow. Could be a bug if it is, so I think we should throw an error.